src/tracing/perfetto/include/matter/tracing/macros_impl.h

```diff
@@ -30,3 +30,10 @@ PERFETTO_DEFINE_CATEGORIES(perfetto::Category("Matter").SetDescription("Matter t
 #define MATTER_TRACE_END(label, group) TRACE_EVENT_END("Matter")
 #define MATTER_TRACE_INSTANT(label, group) TRACE_EVENT_INSTANT("Matter", label, "class_name", group)
 #define MATTER_TRACE_SCOPE(label, group) TRACE_EVENT("Matter", label, "class_name", group)
+
+#define MATTER_TRACE_COUNTER(label, group)                                                                                         \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        static int count##_label = 0;                                                                                              \
+        TRACE_COUNTER("Matter", perfetto::CounterTrack(label), ++count##_label);                                                   \
```

Comment:
Although this also does not have a group, the group has a significant role and may play a significant role in analysis of traces from a particular section , also for filtering out the traces of concern.

Reply:
You are creating overlapping counters ... in this case this is actually wrong: two seemingly different TRACE_COUNTER (because they have different group) will create identical TRACE_COUNTER invocations.

I am unclear what you mean by the statement `the group has a significant role and may play a significant role in analysis of traces from a particular section`. Please give an example. If it is important, it should be contained in the macro invocation to make the macro unique. Otherwise you are intentionally making that different groups generate same macro if the label are the same and I do not understand why that is a desirable behavior.

Reply:
I have removed the group from the counter macro.